core/src/common.cpp

```diff
@@ -362,6 +364,264 @@ namespace irods::http
+	/// \param[in]     _jwk      The JWK containing the required JWA information.
+	/// \param[in]     _alg      The sigining algorithm requested by the signed JWT.
+	///
+	/// \todo Should this be separated out by algorithm familiy?
+	auto add_alg_from_jwk(jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier, const jwt::jwk<jwt::traits::nlohmann_json>& _jwk, std::string_view _alg) -> void {
+		namespace logging = irods::http::log;
+
+		if (_alg == "RS256") {
+			logging::trace("{}: Detected [RS256], attempting extraction of attributes from JWK...", __func__);
+
+			// Get modulus parameter (JWA)
+			auto mod{_jwk.get_jwk_claim("n").as_string()};
+
+			// Get exponent parameter (JWA)
+			auto exp{_jwk.get_jwk_claim("e").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::rs256(jwt::helper::create_public_key_from_rsa_components(mod, exp)));
+		}
+		// For symetric excryption, no parameter 'k' is provided
+		// See OpenID Connect Discovery section 3
+		if (_alg == "HS256") {
+			logging::trace("{}: Detected [HS256], attempting extraction of attributes from JWK...", __func__);
+
+			// Attempt to get key value parameter (JWA)
+			auto key{_jwk.get_jwk_claim("k").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::hs256(key));
+		}
+		if (_alg == "ES256") {
+			logging::trace("{}: Detected [ES256], attempting extraction of attributes from JWK...", __func__);
+
+			// Get curve parameter (JWA)
+			auto crv{_jwk.get_jwk_claim("crv").as_string()};
+
+			// Get x coordinate parameter (JWA)
+			auto x{_jwk.get_jwk_claim("x").as_string()};
+
+			// Get y coordinate parameter (JWA)
+			auto y{_jwk.get_jwk_claim("y").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::es256(jwt::helper::create_public_key_from_ec_components(crv, x, y)));
+		}
+	}
+
+	/// Bleh blah
+	///
+	/// \returns A refernce to the provided jwt::verifier, allowing for chaining.
+	auto add_algorithms_to_verifier(jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier, const jwt::jwks<jwt::traits::nlohmann_json>& _jwks, std::string_view _alg) -> jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& {
+		namespace logging = irods::http::log;
+
+		// Get the JWK the access token was signed with. This is optional.
+		// See RFC 7515 Section 4.1.4
+		try {
+			auto jwk{_jwks.get_jwk(decoded_token.get_key_id())};
+			add_alg_from_jwk(_verifier, jwk, _alg);
+		}
+		catch (const std::runtime_error& e) {
+			// We cannot pick out the specific key used...
+			// Find JWKs that match any of the following:
+			//   - 'alg' (optional) matches exactly
+			//   - 'kty' (required) matches 'alg' family [EC,RSA,oct]
+			//   - 'use' (required if sign & encrypt keys exist) to be 'sig'
+			//
+			// Additionally, symetric algoritms it /seems/ that
+			// "... the octects of the 'client_secret'..." are used for the algorithm...
```

Comment:
Test and verify how exactly the symmetric algorithms work out. Is there a `k` provided or not?

Reply:
`k` is not provided. The algorithm will likely not appear in the JWKs. Key may need to be provided separately.